OneSignalSDK/onesignal/notifications/src/main/java/com/onesignal/notifications/internal/listeners/NotificationListener.kt

```diff
@@ -67,26 +68,30 @@ internal class NotificationListener(
     override suspend fun onNotificationOpened(
         activity: Activity,
         data: JSONArray,
-        notificationId: String,
     ) {
         val config = _configModelStore.model
         val appId: String = config.appId ?: ""
         val subscriptionId: String = _subscriptionManager.subscriptions.push.id
         val deviceType = _deviceService.deviceType
 
         for (i in 0 until data.length()) {
+            val notificationId = NotificationFormatHelper.getOSNotificationIdFromJson(data[i] as JSONObject?)
             if (postedOpenedNotifIds.contains(notificationId)) {
                 continue
             }
-            postedOpenedNotifIds.add(notificationId)
+            if (notificationId != null) {
```

Comment:
Simplification suggestion, we do this `notificationId != null` check twice in this loop, recommend doing an early continue to simplify the code.